vcr/concept/config.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package concept
+
+import (
+	"encoding/json"
+	"regexp"
+	"strings"
+
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/tidwall/gjson"
+)
+
+// Config defines the concept configuration for a VerifiableCredential
+type Config struct {
+	// Concept groups multiple credentials under a single name
+	Concept string `yaml:"concept"`
+	// CredentialType defines the type of the credential. 'VerifiableCredential' is omitted.
+	CredentialType string `yaml:"credentialType"`
+	// Indices contains a set of Index values
+	Indices []Index `yaml:"indices"`
+	// Template is the string template for outputting a credential to a common format
+	// Each <<JSONPath>> value is substituted with the outcome of the JSONPath query
+	Template *string `yaml:"template"`
+}
+
+var templateStringMatcher, _ = regexp.Compile(`<<([a-zA-Z\\.]+)>>`)
+
+func (c Config) transform(vc vc.VerifiableCredential) (Concept, error) {
+	vcBytes, err := json.Marshal(vc)
+	vcString := string(vcBytes)
+	concept := Concept{}
+	var template string
+	if err != nil {
+		return nil, err
+	}
+
+	// if no template is present, we'll return the credential
+	if c.Template == nil {
+		template = vcString
+	} else {
+		template = *c.Template
+
+		// find all << refs >>
+		for _, match := range templateStringMatcher.FindAllStringSubmatch(template, -1) {
```

Comment:
isn't this the same for all credentials of this type? Shouldn't this regexing be done once, when the config is created instead of for every VC being transformed (for performance reasons)?

Reply:
The output is specific per credential. The values between << and >> have to be replaced with the JSONPath result. It's not something you can do genericly